### ISSUE TITLE
from_pretrain with peft adapter on the hub

### ISSUE THREAD

**Issue by glerzing** [lvwerra/trl#379]
Hi, I'm working on the peft integration in trlx. I plan to reuse a few pieces of code if you're ok with it. And when I see this :

![image](https://github.com/lvwerra/trl/assets/47578089/8d2ce899-606e-4280-8bd9-da00133ee19d)

I'm wondering if it shouldn't be this instead : 
![image](https://github.com/lvwerra/trl/assets/47578089/92c4537a-ec64-44f5-a203-400833462121)

Because if the adapter is on the hub, os.path.exists(pretrained_model_name_or_path) should return False I guess.

**Comment by younesbelkada** [comment]
Hi @glerzing 
Sure go ahead you can reuse few pieces of code
I think both conditions leads to the same result (I rapidly tested it - let me know if you see any mistake):
```python
>>> (False and False or True)
True
>>> ((False and False) or True)
True
```
But the proposed changes look cleaner, hence more readable - would you like to open a PR to address this?

**Comment by glerzing** [comment]
Correct me if I am wrong, but there is a subtle difference, here is a simplified version to test the case where the adapter is on the hub but not locally :
``` Python
local_dir_exists = False   # os.path.exists(pretrained_model_name_or_path)
adapter_in_local_dir = False   #"adapter_config.json" in os.listdir(pretrained_model_name_or_path)
file_found_on_the_hub = True   # peft_filename is not None
peft_installed = True   # is_peft_available()

# Equivalent to the current version
if (
    local_dir_exists
    and (adapter_in_local_dir or file_found_on_the_hub is not None)
    and peft_installed
):
    print("case 1 ok")

# Proposed version
if (
    ((local_dir_exists and adapter_in_local_dir)
    or file_found_on_the_hub)
    and peft_installed
):
    print("case 2 ok")
```
It should only output "case 2 ok".

I'll open a PR.

**Comment by younesbelkada** [comment]
Ah you are right! I missed a bracket in the test!
Yes please do open a PR to fix it 💪 !

**Comment by glerzing** [comment]
Solved with #380
